lib/alp_workloads/kvm_workload_utils.pm

```diff
@@ -0,0 +1,201 @@
+use virt_autotest::virtual_network_utils qw(clean_all_virt_networks);
+
+our @EXPORT = qw(
+  set_ct_engine
+  set_kvm_image
+  pull_kvm_container
+  config_host_and_kvm_container
+  cleanup_host_and_kvm_container
+  start_kvm_container
+  enter_kvm_container_sh
+  exit_kvm_container
+  install_tools_within_kvm_container
+  setup_services_within_kvm_container
+  collect_kvm_container_setup_logs
+  setup_kvm_container_from_scratch
+  clean_and_resetup_kvm_container
+);
+
+# Container engine: can be podman(default), docker(not tested)
+our $ct_engine = "podman";
+
+my $DEFAULT_KVM_IMAGE = "registry.opensuse.org/suse/alp/workloads/tumbleweed_containerfiles/suse/alp/workloads/kvm:latest";
+our $kvm_image = $DEFAULT_KVM_IMAGE;
+
+sub set_ct_engine {
+    $ct_engine = shift;
+
+    die "Container engine can only be set to podman or docker!" if ($ct_engine ne "podman" && $ct_engine ne "docker");
+}
+
+sub set_kvm_image {
+    my $_image = shift;
+
+    # Allow customization in case of special test needs
+    if ($_image ne $DEFAULT_KVM_IMAGE) {
+        $kvm_image = $_image;
+        record_info("The kvm container image has been changed from default!", "Now test is using kvm image: $kvm_image.");
+    }
+}
+
+sub pull_kvm_container {
+    # Ensure large enough time is given to download
+    # TODO: uncomment once automation is done
+    # assert_script_run("$ct_engine pull $kvm_image", 1800/get_var('TIMEOUT_SCALE', 1));
+    record_info("Kvm container image is successfully downloaded.", "Image: $kvm_image.");
+
+}
+
+sub config_host_and_kvm_container {
+    # Copy contents from kvm container to host to set it up
+    assert_script_run("clear; $ct_engine container runlabel install $kvm_image");
+    save_screenshot;
+
+    # Replace /etc/kvm-container.conf IMAGE value with $kvm_image if non-default
+    if ($kvm_image ne $DEFAULT_KVM_IMAGE) {
+        assert_script_run("cp /etc/kvm-container.conf /etc/kvm-container.conf.org");
+        assert_script_run(q@sed -i 's#^\s*IMAGE=.*$#IMAGE=@ . $kvm_image . q@#g' /etc/kvm-container.conf@);
+        assert_script_run("grep ^IMAGE= /etc/kvm-container.conf");
+        save_screenshot;
+        record_info("/etc/kvm-container.conf default IMAGE is changed!", "New value is IMAGE=$kvm_image.");
+    }
+
+    # Share /dev/sshserial from host to kvm container for let openqa root-ssh console works
+    assert_script_run("cp /usr/local/bin/kvm-container-manage.sh /usr/local/bin/kvm-container-manage.sh.org");
+    assert_script_run(q@sed -i 's#podman create#podman create --volume /dev/sshserial:/dev/sshserial #g' /usr/local/bin/kvm-container-manage.sh@);
+    script_run("diff /usr/local/bin/kvm-container-manage.sh /usr/local/bin/kvm-container-manage.sh.org");
+    save_screenshot;
+    record_info("kvm-container-manage.sh has been changed to share sol console needed device with kvm container.");
+    assert_script_run("kvm-container-manage.sh create");
+    save_screenshot;
+    record_info("KVM container is created!");
```

Comment:
···
record_info("KVM container is created!", script_output("podman image ls"));
···

Reply:
Debug info in http://10.67.129.185/tests/326#step/setup_kvm_container/32.

Reply:
> exactly there is no checking at all actually before adding script_output

NO, assert_script_run("kvm-container-manage.sh create"), already asserts. Script_output is just output, unless human checks, it does not assert anything. Usually in test, no one would check such record_info outputs, unless test fails. So we should rely on such assert_script_run to automatically checks, rather than human involvement in check. 
But here, I will add the script_output of `$ct_engine ps --all` to show all containers, since it is not given in earlier save_screenshot of the executed script.
BTW, script_run and save_screen , with record_info following that, can show what commands are executed, the output, and notification(record_info) to reviewers, which is usually better than relying only on record_info, in which way, it would not tell you the executed command and usually introduces extra command execution. I only pick this way when the previous script has long outputs which in screenshot, people can not get full output.

Reply:
Adding some sort of output from container engine looks better to test itself now.